src/index.ts

```diff
@@ -290,6 +290,44 @@ class AuthProvider {
     throw ErrorNotInitialized
   }
 
+  /**
+   * A function to to be called before trying to .reconnect()
+   */
+  public canReconnect() {
+    const session = this.iframeWrapper.getSessionID()
+    if (!session) {
+      return false
+    }
+
+    if (session.exp < Date.now()) {
+      return false
+    }
+
+    return true
+  }
+
+  /**
+   * A function to try to reconnect to last login session.
+   * Should be called on event of click function as it opens a popup.
+   */
+  public async reconnect() {
+    const session = this.iframeWrapper.getSessionID()
+    if (session) {
+      if (session.exp < Date.now()) {
+        throw new Error('cannot reconnect, session expired')
+      }
+      const u = new URL(
+        `/v1/reconnect/${this.appId}`,
```

Comment:
Why not let the wallet UI dictate this?

Reply:
Better to have devs do it, if requested we can always switch back to wallet-ui thing. Have expiry here to prevent scenarios where auth opens a tab to reconnect but session is expired.

Reply:
I actually meant letting the wallet UI send the reconnection URL. So we can specialize this later or change the verify app implementation without having to upgrade the Auth SDK.

Reply:
@mmjee updated this to fetch url from wallet